app/models/bulkrax/oai_entry.rb

```diff
@@ -54,6 +56,40 @@ def collections_created?
       end
     end
 
+    # To ensure we capture the correct parse data, we first need to establish the factory_class.
+    # @see https://github.com/samvera-labs/bulkrax/issues/702
+    def establish_factory_class
+      # Why send?  Because in some implementations I've found that folks have privatized the
+      # model_field_mappings method.
+      model_field_names = parser.model_field_mappings
+
+      each_candidate_metadata_node do |node|
+        next unless model_field_names.include?(node.name)
+        add_metadata(node.name, node.content)
+        break
```

Comment:
Am I reading this correctly? 

> 1. Loop through each metadata node, ignoring ones that don't match the configured `model_field_names` 
> 2. Once we hit a `model_field_names` node, add its metadata 
> 3. We have what we need, `break` 

Assuming that's correct, and assuming that there could be multiple `model_field_names` (given that it's plural), wouldn't the `break` mean that we only add the first `model_field_names` and skip all ones after it? 

Example: 
```ruby
model_field_names = ['bar', 'baz']

['foo', 'bar', 'baz'].each do |node|
  next unless model_field_names.include?(node.name) # skip 'foo'
  add_metadata(node.name, node.content) # add 'bar'
  break # skip 'baz'
end
```

Reply:
After discussing with @jeremyf, the more pressing concern related to this is the current behavior of `#factory_class`. I've created [this ticket](https://github.com/samvera-labs/bulkrax/issues/704) capturing that discussion